### DISCUSSION TITLE
Respect git .mailmap

### DISCUSSION THREAD

**Discussion by bagder** [community/community#6754] [Other Feature Feedback, Questions, & Ideas]
git supports a `.mailmap` file that a project can edit to update user information after the fact. It can also be used to mark how two different users (email addresses) is actually the same user.

The GitHub contributor counter doesn't seem to respect this file, which makes some contributors that changed email address over the years, to not have all their contributions counted when listed in the "contributors" view for a repository.

[source link](https://bagder.github.io/github-feedback/mailmap.html)

**Comment by coolreader18** [comment]
I noticed a part of this when looking at a commit.patch url; "usually" .mailmap doesn't matter on GitHub since as long as the email on a commit matches a user's email the commit UI will point to that user. But, I updated my name in GitHub ~recently and even though there's a .mailmap in the repository which fixes the name when viewing patches on the command line, looking at a .patch url shows the old name.

**Comment by bagder** [comment]
It is also very clear in our project (curl). We can easily compare for example the top committers in a `git shortlog -sn` output with the top contributors GitHub shows. When there's a huge discrepancy, it is either this explanation or #6753 happening.

**Comment by n0099** [comment]
https://github.com/orgs/community/discussions/22518
